ros2/pytest_wrapper.py.tpl

```diff
@@ -33,10 +33,12 @@ def finalize_coverage_session(coverage_session: coverage.Coverage) -> None:
 
 
 def main() -> None:
-    test_outputs_dir = os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR')
-    if test_outputs_dir:
-        os.environ['ROS_HOME'] = test_outputs_dir
-        os.environ['ROS_LOG_DIR'] = test_outputs_dir
+    if 'ROS_HOME' not in os.environ and 'ROS_LOG_DIR' not in os.environ:
+        ros_output_dir = os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR')
+        if ros_output_dir is None:
+            ros_output_dir = os.environ.get('TEST_TMPDIR')
+        if ros_output_dir is not None:
```

Comment:
TEST_TMPDIR is always defined, we don't need this `if-`clause here. The previous line should read: `ros_output_dir = os.environ['TEST_TMPDIR']`.

BTW, I would rename ros_output_dir to bazel_test_output_dir for more explicitness.